File: Scripts/newTask/inheritance-single/c-sharp/inheritance-single.cs
using System;

class Animal
{ 
  /* ... */ 
  // ...
  public void speak(){
    Console.WriteLine("Animal speaking");
  }

  public void walk(){
    Console.WriteLine("Animal walkinging");
  }
}
 
class Dog : Animal
{ 
  /* ... */ 
  // ...
  public void speak(){
    Console.WriteLine("Dog barking");
  }
}
 
class Lab : Dog
{ 
  /* ... */ 
  // ...
  public void speak(){
    Console.WriteLine("Lab barking");
  }
}
 
class Collie : Dog
{ 
  /* ... */
  // ... 
  public void speak(){
    Console.WriteLine("Collie barking");
  }
}
 
class Cat : Animal
{ 
  /* ... */
  // ... 
  public void speak(){
    Console.WriteLine("Cat meawing");
  }
}


namespace RosettaCode
{
    class Program
    {
        static void Main(string[] args)
        {
          for (int i=0; i<1000000; ++i) {
           Dog a = new Dog();
            a.speak();
            a.walk();
          }
        }
    }
}

File: Scripts/server/inheritance-single/javascript/inheritance-single.js
function Animal() {
    // ...
}

function Dog() {
    // ...
}

function Lab() {
    // ...
}

function Collie() {
    // ...
}

function Cat() {
    // ...
}

Animal.prototype.speak = function() {console.log("Animal speaking")};
Animal.prototype.walk = function() {console.log("Animal walking")};
Lab.prototype.speak = function() {console.log("Lab barking")};
Collie.prototype.speak = function() {console.log("Collie barking")};
Cat.prototype.speak = function() {console.log("Cat Meawing")};

for ( var i = 0; i < 1000000; i++) {
	Dog.prototype = new Animal();
	Dog.prototype.speak = function() {console.log("Dog barking")}
	var doggy = new Dog();
	doggy.speak();
	doggy.walk();
}
